### PR TITLE
Fix branch name escaping in compare url

### PR DESCRIPTION
**Pull request by M2shad0w**
fix bug #3303
pr chinese name url not decode

### CODE REVIEW

**Comment by lunny** [comment]
@M2shad0w CI failed.

File: modules/context/context.go [hunk 1]
```diff
@@ -103,9 +103,11 @@ func (ctx *Context) Handle(status int, title string, err error) {
 
 	switch status {
 	case 404:
-		ctx.Data["Title"] = "Page Not Found"
+		ctx.Data["Title"] = title
 	case 500:
 		ctx.Data["Title"] = "Internal Server Error"
+	default:
+		ctx.Data["Title"] = title
```
Comment (thehowl) [inline modules/context/context.go]: Why are the changes in this file needed?

File: routers/repo/pull.go [hunk 2]
```diff
@@ -568,7 +569,8 @@ func ParseCompareInfo(ctx *context.Context) (*models.User, *models.Repository, *
 	// format: <base branch>...[<head repo>:]<head branch>
 	// base<-head: master...head:feature
 	// same repo: master...feature
-	infos := strings.Split(ctx.Params("*"), "...")
+	infoPath, _ := url.QueryUnescape(ctx.Params("*"))
```
Comment (thehowl) [inline routers/repo/pull.go]: I'd prefer the error in this case was handled - we can't know for sure that the parameter will be valid. So

```go
infoPath, err := url.QueryUnescape(ctx.Params("*"))
if err != nil {
  ctx.Handle(404, "QueryUnescape", err)
}
```

Comment (M2shad0w) [reply routers/repo/pull.go]: ok

File: modules/context/context.go (hunk 1 above)
Comment (M2shad0w) [reply modules/context/context.go]: i think the status maybe not the num in (404, 500), and then it should be own default value

Comment (thehowl) [reply modules/context/context.go]: I looked through the gitea source - there's no reference anywhere where a code different from 404 or 505 is used. Also - I'm referring to the previous change as well, why did `"Page Not Found"` suffice?

**Comment by M2shad0w** [comment]
merge remote master to my branch ci error
```
2018/01/08 02:12:36 [...user/_obj/gpg_key.go:132 HandleAddGPGKeyError()] [E] AddGPGKey: failed to parse gpg key openpgp: invalid argument: no armored data found
37s
142
2018/01/08 02:12:36 [...user/_obj/gpg_key.go:132 HandleAddGPGKeyError()] [E] AddGPGKey: none of the emails attached to the GPG key could be found: []
```

**Review by lafriks — APPROVED** [review]

**Review by thehowl — APPROVED** [review]

**Pull request merged**